lib/srv/reexec.go

```diff
@@ -682,39 +750,44 @@ func RunForward() (errw io.Writer, code int, err error) {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
-	var buf [1024]byte
-	var fbuf [1]*os.File
 	for {
-		n, fn, err := conn.ReadWithFDs(buf[:], fbuf[:])
+		buf := make([]byte, 0, 1024)
+		fbuf := make([]*os.File, 0, 1)
+		_, fn, err := conn.ReadWithFDs(buf, fbuf)
```

Comment:
This is not going to work, `buf` and `fbuf` aren't getting updated to ever be longer than 0 items, and the `Read` will never fill them to begin with anyway - that's why the tests are now failing.

The previous reading code was totally fine even with the shared buffer, converting the arrays into slices doesn't require changing the rest of the logic.